src/msw/dib.cpp

```diff
@@ -598,17 +638,83 @@ bool wxDIB::Create(const wxImage& image, PixelFormat pf)
 
     // if we have alpha channel, we need to create a 32bpp RGBA DIB, otherwise
     // a 24bpp RGB is sufficient
+    // but use monochrome if requested (to support wxMask)
     const bool hasAlpha = image.HasAlpha();
-    const int bpp = hasAlpha ? 32 : 24;
+    // monochrome DIBs can't express alpha
+    if ( hasAlpha && depth == 1 ) {
+        wxLogError( _("can't convert image with alpha to monochrome") );
+        return false;
+    }
+    const int bpp = depth != -1 ? depth : hasAlpha ? 32 : 24;
 
     if ( !Create(w, h, bpp) )
         return false;
 
+    // convert wxImage's content to monochrome
+    // KLUDGE:  wxScopedPtr doesn't handle T[]
+    unsigned char* eightBitData = nullptr;
+    class Cleaner
+    {
+    public:
+        Cleaner(unsigned char*& p_) :
+            p(p_)
+        {
+        }
+        ~Cleaner()
+        {
+            delete[] p;
+        }
+    private:
+        unsigned char*& p;
+    } cleaner(eightBitData);
+    if ( bpp == 1 )
+    {
+        wxImage quantized;
+        wxPalette* tempPalette;
+        if ( !wxQuantize::Quantize(
+            image,
+            quantized,
+            &tempPalette,
+            2,
+            &eightBitData,
+            wxQUANTIZE_RETURN_8BIT_DATA) )
+        {
+            return false;
+        }
+        wxScopedPtr<wxPalette> palette(tempPalette);
+
+        // try to use palette's colors in result bitmap
+        MemoryHDC hDC;
+        SelectInHDC(hDC, m_handle);
+        RGBQUAD colorTable[256];
+        UINT colors = GetDIBColorTable(hDC, 0, 256, colorTable);
+        for ( UINT i = 0; i < colors; ++i )
+        {
+            if ( !palette->GetRGB(i,
+                                    &colorTable[i].rgbRed,
+                                    &colorTable[i].rgbGreen,
+                                    &colorTable[i].rgbBlue) )
+            {
+                return false;
+            }
+            colorTable[i].rgbReserved = 0;
+        }
+        UINT rc = SetDIBColorTable(hDC, 0, colors, colorTable);
+        if ( rc != colors )
+        {
+            wxLogLastError(wxT("SetDIBColorTable"));
+#if 0   // KLUDGE:  SetDIBColorTable always fails for me,
```

Comment:
What's the last error value? I expect it's something generic and not very useful, but still?